L1Trigger/TrackFindingTracklet/src/PurgeDuplicate.cc

```diff
@@ -108,264 +108,331 @@ void PurgeDuplicate::execute(std::vector<Track>& outputtracks_, unsigned int iSe
+    std::vector<std::vector<std::pair<int, int>>> inputstubidslistsall;
+    std::vector<Tracklet*> inputtrackletsall;
+
+    std::vector<unsigned int> prefTracks;  // Stores all the tracks that are sent to the KF from each bin
+    std::vector<int> prefTrackFit;  // Stores the track seed that corresponds to the associated track in prefTracks
+
+    for (unsigned int bin = 0; bin < settings_.varrinvbins().size() - 1; bin++) {
+      // Get vectors from TrackFit and save them
+      // inputtracklets: Tracklet objects from the FitTrack (not actually fit yet)
+      // inputstublists: L1Stubs for that track
+      // inputstubidslists: Stub stubIDs for that 3rack
+      // mergedstubidslists: the same as inputstubidslists, but will be used during duplicate removal
+      for (unsigned int i = 0; i < inputtrackfits_.size(); i++) {
+        if (inputtrackfits_[i]->nStublists() == 0)
+          continue;
+        if (inputtrackfits_[i]->nStublists() != inputtrackfits_[i]->nTracks())
+          throw "Number of stublists and tracks don't match up!";
+        for (unsigned int j = 0; j < inputtrackfits_[i]->nStublists(); j++) {
+          if (isTrackInBin(findOverlapRInvBins(inputtrackfits_[i]->getTrack(j)), bin)) {
+            if (inputtracklets_.size() > settings_.numtracksperbin())
+              continue;
+            Tracklet* aTrack = inputtrackfits_[i]->getTrack(j);
+            inputtracklets_.push_back(inputtrackfits_[i]->getTrack(j));
+            std::vector<const Stub*> stublist = inputtrackfits_[i]->getStublist(j);
+            inputstublists_.push_back(stublist);
+            std::vector<std::pair<int, int>> stubidslist = inputtrackfits_[i]->getStubidslist(j);
+            inputstubidslists_.push_back(stubidslist);
+            mergedstubidslists_.push_back(stubidslist);
+
+            // Encoding: L1L2=0, L2L3=1, L3L4=2, L5L6=3, D1D2=4, D3D4=5, L1D1=6, L2D1=7
+            // Best Guess:          L1L2 > L1D1 > L2L3 > L2D1 > D1D2 > L3L4 > L5L6 > D3D4
+            // Best Rank:           L1L2 > L3L4 > D3D4 > D1D2 > L2L3 > L2D1 > L5L6 > L1D1
+            // Rank-Informed Guess: L1L2 > L3L4 > L1D1 > L2L3 > L2D1 > D1D2 > L5L6 > D3D4
+            unsigned int curSeed = aTrack->seedIndex();
+            if (curSeed == 0) {
+              seedRank.push_back(1);
+            } else if (curSeed == 2) {
+              seedRank.push_back(2);
+            } else if (curSeed == 5) {
+              seedRank.push_back(3);
+            } else if (curSeed == 4) {
+              seedRank.push_back(4);
+            } else if (curSeed == 1) {
+              seedRank.push_back(5);
+            } else if (curSeed == 7) {
+              seedRank.push_back(6);
+            } else if (curSeed == 3) {
+              seedRank.push_back(7);
+            } else if (curSeed == 6) {
+              seedRank.push_back(8);
+            } else if (settings_.extended()) {
+              seedRank.push_back(9);
```

Comment:
I know you didn't write this Daniel, but I don't understand why this wasn't just done with a single list, i.e.: 
```
std::vector<int> ranks{1,5,2,7,4,3,8,6};
if (settings_.extended()) seedRank.push_back(9);
else seedRank.push_back[ranks[curSeed]];
```
@tomalin maybe you know?

Reply:
I agree your solution is prettier.

Reply:
Should I go ahead and rewrite this section of the code?

Reply:
yes, as Tova suggests.